src/nodes/hypertable_modify.c

```diff
@@ -128,9 +128,14 @@ collect_chunks_from_scan(PlanState *ps, ChunkScanNodes *sn)
 			current_chunk = ts_chunk_get_by_relid(rte->relid, false);
 			if (current_chunk && ts_chunk_is_compressed(current_chunk))
 			{
+				const char *chunk_name;
+				chunk_name = psprintf("%s.%s",
+									  quote_identifier(current_chunk->fd.schema_name.data),
+									  quote_identifier(current_chunk->fd.table_name.data));
+				elog(DEBUG1, "\nChunk being processed: \"%s\"\n", chunk_name);
 				sn->chunks = lappend(sn->chunks, current_chunk);
-				if (ps->plan->qual && !sn->predicates)
-					sn->predicates = ps->plan->qual;
+				/* save filters of this scan node */
+				sn->predicates = lappend(sn->predicates, ps->plan->qual);
```

Comment:
Why are we storing the results of the tree walker in lists and not acting on it immediately once we found a chunk?

Reply:
Fixed.